pkg/streamingpromql/query.go

```diff
@@ -492,6 +574,8 @@ func (q *Query) Close() {
 		types.PutMatrix(v)
 	case promql.Vector:
 		types.VectorPool.Put(v, q.memoryConsumptionTracker)
+	case promql.Scalar:
+		// Nothing to do, we already returned the slice in populateScalarFromScalarOperator.
```

Comment:
do we need to return the slice in the case of a Matrix?

Reply:
Yep - this is already handled a few lines above in this method.